llm_judge.py

```diff
@@ -0,0 +1,102 @@
+from llm import LLM
+from loguru import logger
+import os
+from dotenv import load_dotenv
+import json
+from datetime import datetime
+from pathlib import Path
+from rouge_judge import RougeLJudge
+
+load_dotenv()
+
+
+class BaseJudge:
+    def __init__(self, metric_name, system_prompt):
+        self.metric_name = metric_name
+        self.llm = LLM(
+            temperature=0.2,
+            response_format="json_object",
+            system_prompt=system_prompt,
+            model_name="mistral-small-latest",
+        )
+
+    def evaluate(self, query, result):
+        prompt = f"""
+Given the query: '{query}', evaluate the {self.metric_name} of: '{result}'.
+Provide a numerical score (1-100) and explanation.
+Return json format:
+{{
+    'score': int,
+    'explanation': str
+}}"""
+
+        max_retries = 3
+        for attempt in range(max_retries):
+            try:
+                response = self.llm.complete(prompt)
+                return json.loads(response)
+            except json.JSONDecodeError as e:
+                logger.warning(
+                    f"Attempt {attempt + 1} failed to parse JSON response: {e}"
+                )
+                if attempt == max_retries - 1:
+                    raise ValueError(
+                        f"Failed to get valid JSON after {max_retries} attempts"
+                    )
+                continue
+
+
+class RelevanceJudge(BaseJudge):
```

Comment:
If you want to implement different judges in such a way, create a property maybe in the class? Or even an abstract class with an abstract property?
The implementation using `super().__init__(...)` looks strange